docarray/base_doc/base_node.py

```diff
@@ -20,3 +24,14 @@ def _to_node_protobuf(self) -> 'NodeProto':
         :return: the nested item protobuf message
         """
         ...
+
+    @classmethod
+    @abstractmethod
+    def from_protobuf(cls: Type[T], pb_msg: T) -> T:
+        ...
+
+    def _docarray_to_json_compatible(self):
+        """
+        Convert itself into a json compatible object
+        """
+        ...
```

Comment:
should not it still return self by default ?

Reply:
or maybe not

Reply:
I think it's better to leave this abstract and define for each class, otherwise we say self by default by override it in both IOmixin and BaseDoc class